tests/unit/paginator.test.js

```diff
@@ -28,12 +38,15 @@ describe('Test paginator', () => {
   it('unique IDs do not appear more than once across all pages', (done) => {
     const assertNoDuplicateResults = (assertionVars) => {
       const { paginatedRows } = paginate(rows, assertionVars.page);
-      _.forEach(paginatedRows, ({ ID }) => {
-        assert.isFalse(assertionVars.uIDs.includes(ID));
-        assertionVars.uIDs.push(ID);
+      _.forEach(paginatedRows, ({ id }) => {
+        assert.isFalse(assertionVars.uids.has(id));
+        assertionVars.uids.add(id);
       });
+      if (assertionVars.page.number === assertionVars.totalPages) {
+        assertionVars.uids = new Set();
+      }
     };
-    repeatForAllPages(assertNoDuplicateResults, { uIDs: [] });
+    repeatForAllPages(assertNoDuplicateResults, { uids: new Set() });
```

Comment:
I could be wrong, but you've reset the `uids` at line 45-47 already, right? I guess you only need to reset it once either here or at line 45-47.

Reply:
This `new Set()` initializes `uids` as an empty set so that the first iteration's `uids.has(id)` doesn't fail. The purpose of the `new Set()` in line 46 is to clear this set once we've reached the last page, which is necessary as we start from the beginning with different page sizes.